### DISCUSSION TITLE
STT works in Safari but not in Chrome with Browser-based STT

### DISCUSSION THREAD

**Discussion by schnaker85** [danny-avila/LibreChat#3332] [Troubleshooting] [answered]
### What happened?

We setup STT (Browser-based) on _librechat v0.7.4-rc1_ and it works on safari browser but it does not work on Chrome. 
In Chrome the button is also enabled and can be pressed but after a short time (roughly 5 seconds) it will unselect again and there is no text inserted in the message field. 
I checked the logs (with enabled debug logs) in the browser and backend of librechat but there is nothing present

We also verified that the microphone is enabled/works in chrome. The permission to use the microphone for the librechat url is also given. 

Any ideas where to look or try? What kind of API/javascript library are you using for STT?

It's happening also on different machines. 

### Steps to Reproduce

1. Setup STT on librechat
2. Use Chrome (we use Version 126.0.6478.127) and try to record STT with Browser
3. Text is not present in input field

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by berry-13** [comment]
Hey there,

I think there's a bit of a mix-up between STT and TTS. Since Chrome only uses cloud-based TTS, you'll need to turn on "Use cloud-based browser" in the advanced mode

Is STT working for you? If not, could you give me some more details on how to reproduce the bug? I've tried it myself and it seems to be working fine

**Reply by schnaker85** [reply]
hi @berry-13, 

Sorry for the confusion 🫤 , I  updated the above comment, its about STT (speech-to-text) and not TTS.

As I mentioned above it works all fine in Safari but not in Chrome and our main-browser is Chrome. Do you know if I can enable additional debug/log info about the STT integration (or how to track down the issue better?).
I could provide a screen-recording but it will not show anything additional then the button being clicked.

**Reply by danny-avila** [reply]
hi @schnaker85 

What @berry-13 is saying is that there is an additional option to make this work for Chrome

![image](https://github.com/user-attachments/assets/abeba115-2d0b-41ca-a390-925538f93d85)

**Reply by schnaker85** [reply]
hi @danny-avila, 
Thanks for the additional info, but this is for Text To Speech and we trying to get Speech to Text working in Chrome.

Is there anything I can try/configure to add additional debug/info in Chrome for this integration (Speech to Text).

**Reply by berry-13** [reply]
@schnaker85 sorry, I completely forgot about this issue. I tried STT on Chrome and it works. Could you please provide a screen recording and console logs from the developer tools?

![image](https://github.com/user-attachments/assets/25d17b0c-2d14-433e-bffb-b16d29247fb2)

**Reply by schnaker85** [reply]
hi @berry-13 ,

Here are the two recordings: Chrome which does not work and Safari which works.
Edit: (removed recordings)

**Reply by berry-13** [reply, marked as answer]
hey, @schnaker85. have you tried testing Chrome's STT  on a different website? If that doesn't work, it might be a location issue. You can try checking if it's compatible with your area. I've tested it on multiple Chromium versions and it works perfectly

**Reply by schnaker85** [reply]
Well, we did check on the global demo site of google: https://www.google.com/intl/en/chrome/demos/speech.html and it does not work as well. So its not an issue from librechat but a general chrome or organisation policy issue. 

Thanks for your help anyway!
